### ISSUE TITLE
Cookies returned from get_request_cookies() leave out some properties

### ISSUE THREAD

**Issue by ducaale** [pfernie/cookie_store#16]
After poking around the code, I found the problem lies in `deref()` which returns the `RawCookie` inside `cookie_store::Cookie`
```rust
/// Return an `Iterator` of the cookies for `url` in the store
pub fn get_request_cookies(&self, url: &Url) -> impl Iterator<Item = &RawCookie<'static>> {
    self.matches(url).into_iter().map(|c| c.deref())
}
```

However, `RawCookie` is at that point missing  `path`, `domain`, and `expires` which were left out by `try_from_raw_cookie` as they are already being tracked in `cookie_store::Cookie`
```rust
// These are all tracked via Cookie, clear from RawCookie
let mut builder =
    RawCookieBuilder::new(raw_cookie.name().to_owned(), raw_cookie.value().to_owned());
if let Some(secure) = raw_cookie.secure() {
    builder = builder.secure(secure);
}
if let Some(http_only) = raw_cookie.http_only() {
    builder = builder.http_only(http_only);
}
if let Some(same_site) = raw_cookie.same_site() {
    builder = builder.same_site(same_site);
}
```

I think this also affects `cookie_store::Cookie::path()`, `cookie_store::Cookie::domain()`, and `cookie_store::Cookie::expires()` method calls which are currently delegated to `RawCookie`.

**Comment by pfernie** [comment]
The `get_request_cookies` is (generally) intended for getting cookie values appropriate for sending as part of a `Cookie` header in a request, which should only require the `name` & `value`.

A quick look at your repo suggests you are trying to serialize a subset of cookies related to particular url(s) associated with a session/request? For that sort of functionality, I would have suggested utilizing the `matches` fn, which you appear to be doing in ducaale/xh#125? Is there a problem with this approach; is it problematic to have/serialize a `cookie_store::Cookie` instead of a `RawCookie` (from `cookie-rs`)?

Your comments w.r.t. `path()` et al. are correct, by virtue of the `deref()` to `RawCookie`. In retrospect, exposing that impl is somewhat misleading for the reasons encountered in this issue (incomplete payload).

**Comment by ducaale** [comment]
Thanks for checking. I initially had a trouble with extracting the expiry date from `cookie_store::Cookie` until I later found that I could use `From<Cookie<'a>>` trait implementation for `RawCookie`.

```rust
let cookies = cookie_jar
    .matches(&url)
    .into_iter()
    .map(|c| cookie_crate::Cookie::from(c.clone()))
    .collect();
```

>The get_request_cookies is (generally) intended for getting cookie values appropriate for sending as part of a Cookie header in a request, which should only require the name & value.

This wasn't super clear until I started digging in the code 😅. Maybe this could be mentioned in the docs for `get_request_cookies`?

**Comment by pfernie** [comment]
Right; this is at least a poor documentation situation. I have released a `v0.14.1` that spells this out a little more explicitly, and actually deprecates `get_request_cookies` with a pointer to a new fn `get_request_values` whose signature makes this behavior more explicit.

Thanks for the feedback; I'm closing this issue as solved, but feel free to re-open or open something new if there are still issues with the API.
